get_sample_covariates.py

```diff
@@ -85,35 +79,39 @@ def main(
 
     # sex
     # option 1: separate files
-    # tob
-    tob_sample_qc_ht = hl.read_table(tob_sampleqc_file_path)
-    # convert to pandas
-    tob_sample_qc_df = tob_sample_qc_ht.to_pandas()
-    # extract info and reformat to table
-    tob_sample_qc_df['sample_id'] = [str(s) for s in tob_sample_qc_df['s']]
-    # only retain relevant columns
-    tob_sex_df = tob_sample_qc_df[['sample_id', 'sex']]
-    # bioheart
-    bioheart_sample_qc_ht = hl.read_table(bioheart_sampleqc_file_path)
-    # convert to pandas
-    bioheart_sample_qc_df = bioheart_sample_qc_ht.to_pandas()
-    # extract info and reformat to table
-    bioheart_sample_qc_df['sample_id'] = [str(s) for s in bioheart_sample_qc_df['s']]
-    # only retain relevant columns
-    bioheart_sex_df = bioheart_sample_qc_df[['sample_id', 'sex']]
-    # combine_info
-    sex_df = pd.concat([tob_sex_df, bioheart_sex_df], axis=0)
+    # # tob
+    # tob_sample_qc_ht = hl.read_table(tob_sampleqc_file_path)
+    # # convert to pandas
+    # tob_sample_qc_df = tob_sample_qc_ht.to_pandas()
+    # # extract info and reformat to table
+    # tob_sample_qc_df['sample_id'] = [str(s) for s in tob_sample_qc_df['s']]
+    # # only retain relevant columns
+    # tob_sex_df = tob_sample_qc_df[['sample_id', 'sex']]
+    # # bioheart
+    # bioheart_sample_qc_ht = hl.read_table(bioheart_sampleqc_file_path)
+    # # convert to pandas
+    # bioheart_sample_qc_df = bioheart_sample_qc_ht.to_pandas()
+    # # extract info and reformat to table
+    # bioheart_sample_qc_df['sample_id'] = [str(s) for s in bioheart_sample_qc_df['s']]
+    # # only retain relevant columns
+    # bioheart_sex_df = bioheart_sample_qc_df[['sample_id', 'sex']]
+    # # combine_info
+    # sex_df = pd.concat([tob_sex_df, bioheart_sex_df], axis=0)
 
     # # option 2: combined file
     # at the moment this is not up to date, but ideally this would be what we'd run instead
     # sample_qc_ht_path = dataset_path(f'large_cohort/{vds_version}/sample_qc.ht')
-    # sample_qc_ht = hl.read_table(sample_qc_ht_path)
-    # # convert to pandas
-    # sample_qc_df = sample_qc_ht.to_pandas()
-    # # extract info and reformat to table
-    # sample_qc_df['sample_id'] = [str(s) for s in sample_qc_df['s']]
-    # # only retain relevant columns
-    # sex_df = sample_qc_df[['sample_id', 'sex']]
+    sample_qc_ht = hl.read_table(tenk10k_sampleqc_file_path)
+    # convert to pandas
+    sample_qc_df = sample_qc_ht.to_pandas()
+    # extract info and reformat to table
+    sample_qc_df['sample_id'] = [str(s) for s in sample_qc_df['s']]
+    # go from sex karyotype (XX, XY) to sex numeric coding (2,1)
+    sample_qc_df['sex'] = (
+        sample_qc_df['sex_karyotype'].map({"XY": 1, "XX": 2, "X": 0}).astype(int)
```

Comment:
Looks good for now, but let's come back and generalise this to better capture more complex sex aneuploids.

Reply:
I'll make an issue!

Reply:
https://github.com/populationgenomics/saige-tenk10k/issues/173